src/com/projectkorra/projectkorra/util/DamageHandler.java

```diff
@@ -131,7 +131,7 @@ public static void entityDamageCallback(EntityDamageEvent event) {
 	 * @param entity The entity that is receiving the damage
 	 * @param damage The amount of damage to deal
 	 */
-	public static void damageEntity(final Entity entity, Player source, double damage, final Ability ability, boolean ignoreArmor) {
+	public static void damageEntity(final Entity entity, Player source, double damage, final Ability ability, boolean ignoreArmor, boolean sourcelessDamage) {
```

Comment:
You need to add the old method as a new method, so you don't break addons. Make it just call to itself with `false` as the last parameter.

And why not instead of adding a new parameter, just make `source` null for FireTickTimer, and do 
```java
if (source == null) 
    lent.damage(damage);
else 
    lent.damage(damage, source);
```